tests/test_frame.py

```diff
@@ -2296,9 +2296,7 @@ def test_axes_as_tuple() -> None:
 def test_astype_dict() -> None:
     # GH 447
     df = pd.DataFrame({"a": [1, 2, 3], 43: [4, 5, 6]})
-    columns_types = {"a": "int", 43: "float"}
-    df = df.astype(columns_types)
-    check(assert_type(df, pd.DataFrame), pd.DataFrame)
+    check(assert_type(df.astype({"a": "int", 43: "float"}), pd.DataFrame), pd.DataFrame)
```

Comment:
I like this change to the test, but I would like to also test it the way it was before.

So can you do the test as:
```python
    columns_types = {"a": "int", 43: "float"}
    df = df.astype(columns_types)
    check(assert_type(df, pd.DataFrame), pd.DataFrame)
    check(assert_type(df.astype({"a": "int", 43: "float"}), pd.DataFrame), pd.DataFrame)
```
If this doesn't work, change the `dict` to `Mapping` in `frame.pyi` for `astype()`

Reply:
Actually mypy was failing for ` df = df.astype(columns_types)` even if we changed dict to mapping but I added Any to the list of args so that mypy would not complain

Reply:
To fix this, in `frame.pyi`, change the `dtype` argument to:
```python
dtype:  AstypeArg | Mapping[Any,  Dtype] | Series
```
Apparently, we can't use `AstypeArg` if someone stores the types in an untyped `dict`